src/tests/fakes/fake_input.rs

```diff
@@ -135,13 +135,10 @@ pub fn get_interface() -> NetworkInterface {
     }
 }
 
-pub fn create_fake_lookup_addr(
+pub fn create_fake_resolver_factory(
```

Comment:
The honest truth is that this is my biggest concern with this patch. This means that the whole logic in resolver.rs is not tested. Previously when using the dns_queue, what we did was to pass just the dns lookup function itself and use the queue with it. Do you think we can do something similar here?

Reply:
You're definitely right! I will try to encapsulate the trust-dns `AsyncResolver` in a struct that will be injected in the `Resolver` and get rid of the `FakeResolver`; this way the "delivery" of the requests can actually be tested.